crates/holochain/src/core/state/cascade.rs

```diff
@@ -441,42 +565,63 @@ where
         }
     }
 
-    fn valid_header(&self, hash: &HeaderHash) -> CascadeResult<bool> {
-        Ok(self.meta_vault.has_registered_store_element(&hash)?
-            || self.meta_cache.has_registered_store_element(&hash)?)
+    /// Check if this hash has been validated.
+    /// Elements can end up in the cache or integrated table because
+    /// they were gossiped to you or you authored them.
+    /// If you care about the hash you are using being valid in the same
+    /// way as if you got it from the StoreElement authority you can use
+    /// this function to verify that constraint.
+    ///
+    /// An example of how this could go wrong is you do a get for a HeaderHash
+    /// where you are the authority for the RegisterAgentActivity for this header.
+    /// That hash is in your integrated db so you find it but the element has failed
+    /// app validation. The header appears valid even though it isn't because as a
+    /// RegisterAgentActivity authority you haven't run app validation.
```

Comment:
nice docs

Reply:
This is what I meant today about the comment that could be a test :laughing: